core/ballerina/encoders.bal

```diff
@@ -0,0 +1,242 @@
+isolated function encodeDataset(Dataset dataset, TransferSyntax transferSyntax,
+        boolean encodeSorted) returns byte[]|EncodingError {
+    Dataset encodingDataset = dataset;
+    do {
+        byte[] datasetBytes = [];
+
+        if encodeSorted {
+            Dataset sortedDataset = getSortedDataset(dataset);
+            encodingDataset = sortedDataset;
+        }
+
+        foreach DataElement dataElement in encodingDataset {
+            byte[] dataElementBytes = check encodeDataElement(dataElement, transferSyntax);
+            datasetBytes.push(...dataElementBytes);
+        }
+
+        return datasetBytes;
+    } on fail error e {
+        return error EncodingError(string `Data set encoding failed: ${dataset.toString()}`, e);
+    }
+}
+
+isolated function encodeDataElement(DataElement dataElement,
```

Comment:
Add bal docs and comments to all the funtions, utilities

Reply:
Added Bal doc comments to all private functions as well in https://github.com/ballerina-platform/module-ballerinax-health.dicom/commit/3807383514d097fe372c936aa46ef84a627a58d3